### PR TITLE
harden flacky test: assert graceful up to 10 seconds for the async

### PR DESCRIPTION
**Pull request by ivy-rew**
fix and leave some teaching notes :)

### CODE REVIEW

File: connectivity/connectivity-demos-test/src_test/rest/bpm/TestAsyncInvocation.java [hunk 1]
```diff
@@ -3,14 +3,22 @@
 import static org.assertj.core.api.Assertions.assertThat;
 
 import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Supplier;
 
+import org.awaitility.Awaitility;
```
Comment (ivy-rew) [inline connectivity/connectivity-demos-test/src_test/rest/bpm/TestAsyncInvocation.java:11]: number one java library to do assertions with a timeout ... without having to implement retry and timeout checking on your own. :nerd_face:

File: connectivity/connectivity-demos-test/src_test/rest/bpm/TestAsyncInvocation.java [hunk 2]
```diff
@@ -22,12 +30,17 @@ void callBatchAsync(BpmClient bpmClient) throws InterruptedException
     ExecutionResult result = bpmClient.start().process("rest/asyncInvocation/callBatchAsync.ivp").execute();
     assertThat(result).isNotNull();
     
-    Thread.sleep(4000);
```
Comment (ivy-rew) [inline connectivity/connectivity-demos-test/src_test/rest/bpm/TestAsyncInvocation.java:25]: get rid of this bad practice; to have strict timeouts ...  the process is being delayed by design for 3 secs; but we can't assume how fast the firing of the signal will be available ... Q.E.D in our frequently failing pipeline in the night. :watch:

File: connectivity/connectivity-demos-test/src_test/rest/bpm/TestAsyncInvocation.java [hunk 3]
```diff
@@ -22,12 +30,17 @@ void callBatchAsync(BpmClient bpmClient) throws InterruptedException
...
+    
+    Supplier<List<String>> signals = () -> new ArrayList<>(query.executor().results())
+      .stream().map(ISignalEvent::getSignalCode).map(SignalCode::toString)
+      .toList();
+    Awaitility.await()
+      .atLeast(3, TimeUnit.SECONDS) // default delay: on process being called
+      .atMost(10, TimeUnit.SECONDS)
+      .until(() -> signals.get(), 
+        codes -> !codes.isEmpty() && Objects.equals(codes.get(0), "rest:batch:result:ok")
+      );
```
Comment (ivy-rew) [inline connectivity/connectivity-demos-test/src_test/rest/bpm/TestAsyncInvocation.java:43]: now it's easy and apparent, that we are dealing with timing issues here :nerd_face:

**Comment by ivy-rew** [comment]
Did you check this? Any questions? @ivy-edp

**Review by ivy-edp — APPROVED** [review]
👍

**Pull request merged**